opentelemetry-stdout/src/logs/exporter.rs

```diff
@@ -1,38 +1,24 @@
 use async_trait::async_trait;
+use chrono::{DateTime, Utc};
 use core::fmt;
-use opentelemetry::{
-    logs::{LogError, LogResult},
-    ExportError,
-};
 use opentelemetry_sdk::export::logs::{ExportResult, LogData};
 use opentelemetry_sdk::Resource;
-use std::borrow::Cow;
-use std::io::{stdout, Write};
+use std::{borrow::Cow, sync::atomic};
 
-type Encoder =
-    Box<dyn Fn(&mut dyn Write, crate::logs::transform::LogData) -> LogResult<()> + Send + Sync>;
-
-/// A [`LogExporter`] that writes to [`Stdout`] or other configured [`Write`].
-///
-/// [`LogExporter`]: opentelemetry_sdk::export::logs::LogExporter
-/// [`Write`]: std::io::Write
-/// [`Stdout`]: std::io::Stdout
+/// An OpenTelemetry exporter that writes Logs to stdout on export.
 pub struct LogExporter {
-    writer: Option<Box<dyn Write + Send + Sync>>,
-    encoder: Encoder,
     resource: Resource,
-}
-
-impl LogExporter {
-    /// Create a builder to configure this exporter.
-    pub fn builder() -> LogExporterBuilder {
-        Default::default()
-    }
+    is_shutdown: atomic::AtomicBool,
+    resource_emitted: bool,
```

Comment:
Similar to `is_shutdown`, `resource_emitted` needs thread synchronization as well?

Reply:
Do we? Processor ensures export() calls are never done concurrently.

Reply:
Ah yes. I was wondering about the SimpleProcessor case, but we take a Mutex there anyway.